### DISCUSSION TITLE
SG-Ready zur Ansteuerung einer Wärmepumpe konfigurieren (Test vor realem Einsatz)

### DISCUSSION THREAD

**Discussion by BikerGrisu** [camueller/SmartApplianceEnabler#276] [Allgemein]
Hallo,
wir erhalten in den nächsten Wochen eine Daikin Alltherma Wärmepumpe. Diese hat das SG-Ready Label. Deshalb habe ich jetzt nach einer Möglichkeit gesucht, dass die Wärmepumpe sich an unserem überschüssigen PV-Strom "bedient". Wäre ja zu schade, wenn der nur eingespeist wird.
Der SmartApplianceEnabler scheint mir da sehr erfolgversprechend zu sein. Bevor ich jetzt davon ausgehe, dass das schon irgendwie funktionieren wird und ich meinem Elektriker mitteile, dass er noch Platz für einen Raspberry PI auf Hutschiene und alles was da sonst noch so dazugehört (z. Bsp. 2 Relais für die potentialfreie Ansteuerung der WP), vorsehen soll, würde ich das prinzipiell gerne vorab einmal testen. Nur weiss ich nicht so recht, was ich da alles konfigurieren muss?
Meine Umgebung:
- SMA Homemanager
- Raspberry PI
- S0 Leistungsmesser (geplant)
- Daikin Alltherma
- evtl. FritzBox als "vor Ort" LAN in der Nähe der WP

Die Daikin WP bietet eine SG-Ready Schnittstelle mit 2 potentialfreien Kontakten. Ein Kontakt ist mit SG-Ready, der andere mit EVU bezeichnet. 
Den Raspberry PI habe ich bereits schon mit dem Linux OS und dem SmartApplianceEnabler ausgestattet. Zu meiner Überraschung wurde eine testhalber konfigurierte Wärmepumpe sogar vom SMA Homemanager erkannt. Ich hatte mich noch gefragt, wie man dem Homemanager denn mitteilt, dass da ein neues Gerät ist, aber das scheint mit dem SEMP-Protokoll sehr gut automatisch zu funktionieren. Super!
Jetzt stellt sich mir halt die Frage, was ich am SmartApplianceEnabler am Gerät "Wärmepumpe" konfigurieren muss, damit der Homemanager via SG-Ready Funktion 2 GPIOs am Raspberry ansteuert?
Und vielleicht weiss auch noch jemand, wie ich die Funktion gezielt zum testen im Homemanager auslösen kann. In Ermangelung eines Oszilloskops um damit den ganzen Nachmittag auf eine steigende Flanke an einem GPIO des Raspberries zu triggern. Da wäre ein DMM momentan die bevorzugte Variante und dann am besten ein Klick im Homemanager zum testen.

Vielen Dank, Frank

PS: Oder sollte ich vielleicht die mir vorschwebende "Erfolgsprämie" gleich in den persönlichen Support des Entwicklers investieren?

**Comment by camueller** [comment]
Der EVU-Kontakt ist (vermutlich) nur dafür da, damit der Stromversorger die WP bei Bedarf zu Spitzenzeiten abschalten kann, d.h. dieser Kontakt ist für Dein Vorhaben nicht relevant.

Für den SG-Ready-Kontakt benötigst Du ein Solid State Relais (das hier habe ich im EInsatz: https://www.pollin.de/p/solid-state-relais-xssr-da2420-3-32-v-20-a-240-v-340470 - sieht bei mir so aus https://github.com/camueller/SmartApplianceEnabler/blob/master/pics/Schaltschrank.png, wobei Du den Kühlkörper nicht brauchst). Das Solid State Relais wird an den Raspberry Pi angeschlossen - siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/GPIOSwitch_DE.md .

Auch der S0-Zähler (sollte mind. 1000 Impulse/kWh haben!) wird direkt an den Raspberry Pi angeschlossen (Pull-Up-Verdrahtung!): https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/S0Meter_DE.md

Im Smart Appliance Enabler musst Du für das Gerät einen Zeitplan mit Überschussenergie konfigurieren: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Schedules_DE.md

Warum willst Du in eine Fritzbox als "vor Ort" LAN in der Nähe der WP platzieren? Wenn Du dort bereits LAN hast, solltest Du das direkt für den Raspberry Pi verwenden. Falls nicht, hilft die die Fritzbox auch nicht weiter. Minimal brauchst Du stabiles WLAN im Bereich der WP - ich würde Dir aber eher zu Ethernet bzw. LAN raten. Ich habe damals auch extra meine Ethernet-Verkabelung bis in den Heizungskeller erweitert.

**Reply by BikerGrisu** [reply]
Hallo,

und schon mal vielen, vielen Dank für die schnelle Reaktion. :-)

> Der EVU-Kontakt ist (vermutlich) nur dafür da, damit der Stromversorger die WP bei Bedarf zu Spitzenzeiten abschalten kann, d.h. dieser Kontakt ist für Dein Vorhaben nicht relevant.
> 

Das wird durchaus so sein. Zumal der EVU-Kontakt mit dem Signal HT/NT belegt werden soll. Das ist dann sicherlich die Geschichte, wenn der Energieversorger mittels Rundsteuersignal die Wärmepumpe steuern möchte. Ist bei mir aber nicht der Fall.
Der Anschlussplan der Daikin WP zeigt jedoch das "von Außen" das Signal SG1 auf einen Kontakt mit der Bezeichnung "SmartGrid" und das Signal SG2 auf den Kontakt mit der Bezeichnung "EVU" gelegt werden soll. Wenn dann in der WP die SG Funktion aktiviert wird, wird automatisch die Funktion HT/NT deaktiviert.
D. h. dann aber auch, dass der Homemanager letztendlich idealerweise 2 GPIOs des Raspberries ansteuern sollte. Und das gilt es jetzt herauszufinden, wie und ob das funktioniert. Und ob das überhaupt erforderlich ist und ein SG-Ready Kontakt vielleicht schon ausreicht.


> Für den SG-Ready-Kontakt benötigst Du ein Solid State Relais (das hier habe ich im EInsatz: https://www.pollin.de/p/solid-state-relais-xssr-da2420-3-32-v-20-a-240-v-340470 - sieht bei mir so aus https://github.com/camueller/SmartApplianceEnabler/blob/master/pics/Schaltschrank.png, wobei Du den Kühlkörper nicht brauchst). Das Solid State Relais wird an den Raspberry Pi angeschlossen - siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/GPIOSwitch_DE.md .
> 

Hmm, ich hatte da bislang zwei stinknormale Relais im Kopf, aber ich schaue mir den Vorschlag auch mal gerne an. Ich muss dann halt nur schauen, wo der Unterschied in den Kosten liegt, was evtl. einfacher zu handhaben ist und zur Zeit spielt dann wohl auch die Verfügbarkeit keine so kleine Rolle.

> Auch der S0-Zähler (sollte mind. 1000 Impulse/kWh haben!) wird direkt an den Raspberry Pi angeschlossen (Pull-Up-Verdrahtung!): https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/S0Meter_DE.md
> 

Ja, die von Dir vorgeschlagenen Geräte gefallen mir. Der Tipp ist klasse.

> Im Smart Appliance Enabler musst Du für das Gerät einen Zeitplan mit Überschussenergie konfigurieren: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Schedules_DE.md
> 

OK, da schaue ich noch mal etwas genauer hin. Für meinen ersten Test bin ich da wohl etwas flüchtig drüber gegangen. Ich wollte nur erstmal das Prinzip verstanden haben.

> Warum willst Du in eine Fritzbox als "vor Ort" LAN in der Nähe der WP platzieren? Wenn Du dort bereits LAN hast, solltest Du das direkt für den Raspberry Pi verwenden. Falls nicht, hilft die die Fritzbox auch nicht weiter. Minimal brauchst Du stabiles WLAN im Bereich der WP - ich würde Dir aber eher zu Ethernet bzw. LAN raten. Ich habe damals auch extra meine Ethernet-Verkabelung bis in den Heizungskeller erweitert.

Der Elektriker soll zunächst mal ein CAT-7 Kabel zusammen mit der erforderlichen neuen Elektrik in den Heizungsraum verlegen. Momentan muss ich dann wohl oder übel die noch rumliegende FritzBox für ein WLAN nutzen, weil ich gerade keinen Raspberry mit LAN Anschluss hier zur Verfügung habe. Und die Dinger sind wohl leider momentan auch nicht so zahlreich bei den gängigen Distris verfügbar. Ich habe aber auch nur mal eben schnell bei Reichelt vorbeigeschaut.
Idealerweise wären im Heizungsraum natürlich alle Geräte die ins Hausnetz sollen auch per LAN angebunden. Da bin ich auch noch am überlegen, ob statt des S0-Leistungszählers nicht ein HTTP-Zähler einfacher wäre (der müsste nämlich nicht an den Raspberry angeschlossen werden). Für die Verteilung des LANs nimmt man natürlich standardmässig nen Switch. Aber da ich noch eine Fritzbox rumliegen habe, könnte ich mir den evtl. sparen.

Gruss, Frank

**Comment by Trunks1982** [comment]
Kannst auch nen Shelly 1 nehmen dieser ist potenzialfrei.

**Reply by BikerGrisu** [reply]
Hallo,

ja, diese Shelly Plugs sind tatsächlich eine coole Sache für den Smarthome Bereich.
Aber in meinem Anwendungsfall, glaube ich, bin ich mit dem Raspberry besser unterwegs. Der hat die GPIOs an Bord. Da eine Lochrasterplatine mit Pfostenleiste draufstecken und 2 5V Relais drauflöten halte ich momentan für etwas einfacher, als 2 Shelly 1 so mit dem SmartApplianceEnabler zu verheiraten, dass am Ende diese SG-Ready Funktion dabei herauskommt.

Aber trotzdem vielen Dank, für den Hinweis.

Gruss, Frank

**Reply by camueller** [reply]
Mit Relais meinst Du hoffentlich ein Solid State Relais. Ein klassisches Relais mit Spule zieht zuviel Strom für den Raspberry Pi.

**Reply by BikerGrisu** [reply]
> Mit Relais meinst Du hoffentlich ein Solid State Relais. Ein klassisches Relais mit Spule zieht zuviel Strom für den Raspberry Pi.

Moin,

danke für den Hinweis. Soweit im voraus hatte ich noch nicht gedacht. Ich hatte die Erwartungshaltung, dass die GPIOs des Raspberries wohl in der Lage seien die paar mA für ein kleines Relais zu treiben. Aber tatsächlich könnte das sogar schon bei den Relais, die ich im Kopf habe, grenzwertig sein. Ok, dann kommt eben noch so etwas wie der gute alte ULN2004 zum Einsatz. Das macht den Kohl jetzt auch nicht mehr fett. :-)

Aber vielleicht noch mal etwas zum SAE. Wenn ich das richtig sehe, ich habe mir als Gerät eine Wärmepumpe angelegt, kann dieses Gerät "nur" einen Schaltkontakt schalten. D. h. im Endeffekt müsste ich einen SG-Ready Kontakt auf einen statischen Pegel legen und den anderen SG-Ready Kontakt kann ich dann aktiv beeinflussen.

Gruss, Frank

**Reply by camueller** [reply]
Der SAE schaltet ein Gerät, indem der einen GPIO des Raspberry Pi auf high oder low setzt. Was Du daraus machst, musst Du entscheiden :-)

**Reply by premultiply** [reply]
Die typischen Relaiskarten haben entsprechende Treiber-Transistoren usw. drauf so dass man sie direkt an die GPIOs anschließen kann.
Bekommt man im Netz quasi hinterhergeworfen.

**Reply by camueller** [reply]
Stimmt, so ein 8-fach Relaismoduls verwende ich für meine Sprinkler-Anlage im Garten (https://github.com/camueller/OpenSprinkler-Firmware). Gibt es hier günstiger in grosser Auswahl: https://www.amazon.de/s?k=relaismodul&

**Reply by Lausbua** [reply]
Ich hatte damals nach einer Lösung gesucht um das ganze auf Hutschiene vernünftig unterzubringen und kann auch diese Relaisplatine empfehlen:

https://www.reichelt.de/raspberry-pi-relais-modul-8-channel-hutschiene-hls8l-dc5v-s-rpi-relais-8ch-p253984.html

Die Spannungsversorgung erfolgt mit einem zusätzlichen 5V Hutschienennetzteil.
Die Versorungsspannung wird direkt an die Platine angeschlossen (5V/GND).
Den Rapsi aufstecken und fertig.
Die Verbindung zu den Relais/GPIO sowie die Spannungsversorgerung vom Raspi erfolgt direkt über den Raspi-Sockel.
Unter anderem sind die Spulen der Relai eigens mit den 5V versorgt, die Schaltvorgänge laufen über einen Optokoppler.
